### ISSUE TITLE
Native Apple Silicon Support

### ISSUE THREAD

**Issue by fishcharlie** [spesmilo/electrum#7557]
Electrum should release a Universal or separate Apple Silicon build of the application. This would allow users to use the application with increased performance without having to rely on Rosetta 2.

**Comment by SomberNight** [comment]
1. This might be non-trivial to do with our current scripts for codesigning and being able to strip the codesign signatures to verify reproducible builds.
2. Not sure if pyinstaller supports this yet.
    - EDIT(2026-04): looks like it does since https://github.com/pyinstaller/pyinstaller/pull/5581 
3. Looks like one needs xcode 12.2 to build universal2 binaries, which requires macos 10.15 for the build system, which based on past experience would mean bumping the min supported macos version for users to the ~same. (Currently we support 10.13+)
    - EDIT(2026-04): Since https://github.com/spesmilo/electrum/commit/86f6e3dbcb23e5c232655adf5a5f34b68e8dede2, the release build system uses macOS 11.7.10 and xcode 13.2.  We could bump this further a bit if that helped.
4. Note that we would want to build on amd64.
    - that is, ether building a universal2 binary, or two separate x86_64 and arm64 binaries would be fine, however if at all possible the build host machine should stay x86_64

Contributions are welcome.

**Comment by ballo** [comment]
Why was this closed? Apple hasn't sold Intel Macs in a long while

**Comment by ballo** [comment]
pyinstaller can build native binaries BTW
https://github.com/pyinstaller/pyinstaller/issues/5315

**Comment by Ishaanahuja7** [comment]
Any update on this? Qt 5.15.9 community version also supports building on apple silicon now natively. Official Qt download link [here](https://download.qt.io/official_releases/qt/5.15/5.15.9/single)

**Comment by Ishaanahuja7** [comment]
@SomberNight

**Comment by Ishaanahuja7** [comment]
Yes, arm64 can be built from amd64 mac. But min mac os then needs to be updated to 10.15+ and which requires xcode 12.2. Builds will be reproducible as in the amd64 mac case so that shouldn't be an issue.

**Comment by Ishaanahuja7** [comment]
Any updates on this yet? @SomberNight

**Comment by SomberNight** [comment]
We will be moving the build system for the mac binary to a new VM (macOS 10.14.6, xcode 11.3.1 -> macOS 11.7.10, xcode 13.2). See https://github.com/spesmilo/electrum/pull/8738
This itself is unrelated to arm64.

However, as part of that change, we accidentally bumped into a reproducibility failure that was caused precisely due to compiling some .so files as universal (x86_64+arm64) instead of just x86_64. We had to specifically disable that, to make it reproducible.
See https://github.com/spesmilo/electrum/pull/8738/commits/77f84060f459af7e8b274ef1be5dbb65298e6901

> Any updates on this yet?

Please take a look at that commit and try to make hid.so build reproducibly as a universal binary.
Pull request welcome.

**Comment by Ishaanahuja7** [comment]
Thanks a lot for adding initial support for apple silicon macs @SomberNight. According to @ecdsa, hid.cpython-310-darwin.so is reproducible https://github.com/spesmilo/electrum/pull/8738#issuecomment-1851821524. Can you enable universal builds now?

**Comment by ecdsa** [comment]
@Ishaanahuja7 No, I wrote that this file is reproducible **because** the native support has been disabled.

**Comment by Ishaanahuja7** [comment]
@ecdsa Sorry my bad

**Comment by Ishaanahuja7** [comment]
Any update on this? @ecdsa @SomberNight

**Comment by SomberNight** [comment]
As far as I know, no one spent time on this. You or anyone else is welcome to try changing the build scripts and opening a PR once it works.

**Comment by superbonaci** [comment]
It would be important to have a look at this, but the project has lots of dependencies that need to run too on ARM64.

**Comment by f321x** [comment]
From MacOS 28 onwards Apple is planning to offer Rosetta only for older games, https://support.apple.com/en-us/102527.
Apparently the current [MacOS version 26.4 already warns users](https://appleinsider.com/articles/26/02/16/macos-tahoe-264-warns-if-your-apps-wont-work-when-rosetta-2-dies) that an x86_64 app will stop working in the future.

It seems like they release a new version roughly once per year, which would mean MacOS 28 can be expected somewhere around September 2027. 
So I assume this would be kind of a deadline for this issue if we want to support MacOS 28 from the beginning.

Maybe we can cross-compile the compiled dependencies to arm64 and use the pyinstaller [`--target-architecture` argument](https://pyinstaller.org/en/stable/usage.html#cmdoption-target-architecture) to bundle Electrum with an arm64 CPython build and the arm64 MacOS shared libraries inside the existing x86_64 MacOS VM currently used for building.

**Comment by marcuslannister** [comment]
Hi All,

I have forked and built an Arm version here, you can try it:

[download](https://github.com/marcuslannister/electrum/releases/tag/macos-arm64-test-f51b2ae3c)

**Comment by Zettt** [comment]
@marcuslannister why not send a pull request?

**Comment by marcuslannister** [comment]
> [@marcuslannister](https://github.com/marcuslannister) why not send a pull request?

I want someone to help me test it before I send a PR.

**Comment by ecdsa** [comment]
are the builds reproducible?

**Comment by marcuslannister** [comment]
> are the builds reproducible?

Yes, You can check on my fork. [link](https://github.com/marcuslannister/electrum)

**Comment by SomberNight** [comment]
> I have forked and built an Arm version here, you can try it:

You forgot to tell your slop-generator that we would want to build from x86_64, so e.g. cross-compile where needed.
That was point 4 in the [first reply](https://github.com/spesmilo/electrum/issues/7557#issuecomment-963440741) in the thread, but it was [mentioned](https://github.com/spesmilo/electrum/issues/7557#issuecomment-1587383992) [multiple](https://github.com/spesmilo/electrum/issues/7557#issuecomment-1592504062) [times](https://github.com/spesmilo/electrum/issues/7557#issuecomment-3945364799).

```
vagrant@users-iMac-Pro electrum % ELECTRUM_MACOS_ARCH=arm64 ./contrib/osx/make_osx.sh 
💬 INFO:  Found 8 CPUs, which we might use for building.
host architecture 'x86_64' cannot build native arm64; run on Apple Silicon
🗯 ERROR: host cannot build native arm64
```
-----

Either building a universal2 binary, or two separate x86_64 and arm64 binaries would be fine, however if at all possible the build host machine should stay x86_64.  That way my Linux x86_64 machine, which is very much capable of running macOS inside QEMU/KVM (x86_64) would still be able to build binaries and compare those against the official releases for reproducibility.

**Comment by SomberNight** [comment]
> I have forked and built an Arm version here, you can try it:

Also, did you test codesigning+notarization or at least just codesigning the binary?
As I said in point 1 in the [first reply](https://github.com/spesmilo/electrum/issues/7557#issuecomment-963440741) in the thread, that is one of the expected difficulties in this whole endeavour. This is because I suspect what we are doing in the [`extract_sigs.sh`](https://github.com/spesmilo/electrum/blob/f60cdb0f405cb4ae8511d2bc9cfd4beab8a894bb/contrib/osx/extract_sigs.sh) and [`apply_sigs.sh`](https://github.com/spesmilo/electrum/blob/f60cdb0f405cb4ae8511d2bc9cfd4beab8a894bb/contrib/osx/apply_sigs.sh) scripts might not work for universal2 or even arm64 mach-o files.

-----

>> are the builds reproducible?
>
> Yes, You can check on my fork. [link](https://github.com/marcuslannister/electrum)

What exactly works and is reproducible? Do you mean building an arm64-only binary from an arm64 host, without codesigning, without notarization, works and is reproducible? That's a start, but it is not something we can release.

**Comment by ssotomayor** [comment]
Spent some time on this and made #10745, basically with the idea of @f321x  https://github.com/spesmilo/electrum/issues/7557#issuecomment-3945364799, the result is a cross-compiled arm64 build that can be built on Big Sur (OSX 11.7.11), tested codesigning and building couple times got hash `95be818c1b97620e6c3981d08cba0eea1bad6e2b9d0b49680de9577931157509` although i had to install some build tools with MacPorts.

What i couldn't test was lightning and the QR Scanner. Seems to work just fine on my M4. I don't have an Intel mac at hand so went with a hackintosh to test the x86 cross-build. Hopefully goes fine.

**Comment by CicerBro** [comment]
Quite surprised this still doesn't exist in 2026. I'll probably have to move to Sparrow Wallet or Blockstream Green or something similar.
